datafusion-cli/src/print_format.rs

```diff
@@ -161,7 +161,7 @@ impl PrintFormat {
         maxrows: MaxRows,
         with_header: bool,
     ) -> Result<()> {
-        if batches.is_empty() || batches[0].num_rows() == 0 {
+        if batches.is_empty() {
```

Comment:
This is the fix (thanks @Jefffrey for finding this). The rest of the PR is tests

Reply:
To keep the intended behavior, I think this should probably be:

```suggestion
        if batches.is_empty() || batches.iter().all(|b| b.num_rows() == 0) {
```

However, I think the intention is wrong. We should totally print the header even if there's no data. This entire "don't print anything if the result is empty" is super confusing, at least in my opinion.

Reply:
> However, I think the intention is wrong. We should totally print the header even if there's no data. This entire "don't print anything if the result is empty" is super confusing, at least in my opinion.

This is a good call -- I will update and add a test

I believe the logic is to support generating output incrementally ("Streaming") rather than waiting for the entire output to be generated as `Vec<RecordBatch>`. 

However, that logic is sort of split between this module and `print_format.rs` which I agree is confusing

cc @berkaysynnada

Reply:
fixed in 2ef1f9dcd